sdk/python/smithy/dbmanagers/remote_db_manager.py

```diff
@@ -0,0 +1,207 @@
+        :param max_backoff_seconds: The maximum backoff time for retrying connection
+        :type max_backoff_seconds: str
+        :param backoff_multiplier: The multiplier for the backoff time.
+        :type backoff_multiplier: float
+        :param page_size: The number of findings to retrieve per page. Default is 100
+        :type page_size: int
+        :param logger: An instance of the Logger class for logging. If not provided, a default logger will be used.
+        :type logger: Optional[Logger]
+        :raises TypeError: If the logger is not an instance of Logger or None.
+        """
+
+        super().__init__(instance_id, logger)
+
+        self.findings_address = findings_address
+        self.max_attempts = max_attempts
+        self.initial_backoff_seconds = initial_backoff_seconds
+        self.max_backoff_seconds = max_backoff_seconds
+        self.backoff_multiplier = backoff_multiplier
+        self.page_size = page_size
+
+        self._channel = grpc.insecure_channel(self.findings_address)
+
+        self.stub = FindingsServiceStub(self._channel)
+
+    @override
+    def get_findings(
+        self, page_num: Optional[int] = None, page_size: Optional[int] = None
+    ) -> List[Finding]:
+        """
+        Retrieves **all** findings from the remote database (unless `page_num` is used) which are associated to the class variable `instance_id`.
+
+        :param page_num: Optional parameter to specify the page number for pagination. If not provided, all findings will be retrieved. Otherwise it will retrieve the `SMITHY_REMOTE_CLIENT_PAGE_SIZE`(default 100) findings for the given page number.
+        :type page_num: Optional[int]
+
+        :param page_size: Optional parameter to specify the number of findings per page. If not provided, the page size specified by the ENV variable `SMITHY_REMOTE_CLIENT_PAGE_SIZE` will be used (default is 100).
+        :type page_size: Optional[int]
+
+        :return: A list of findings retrieved from the remote database.
+        """
+
+        page = 0
+
+        get_findings_page_size = self.page_size
+        if page_num and isinstance(page_num, int) and page_num >= 0:
+            page = page_num
+
+        if page_size and isinstance(page_size, int) and page_size > 0:
+            get_findings_page_size = page_size
+
+        if page_size and not page_num:
+            self._log.warning(
+                "page_size is provided but page_num is not. Defaulting to page_num 0."
+            )
+
+        all_findings = []
+        try:
+            while True:
+                resp = self.stub.GetFindings(
+                    GetFindingsRequest(
+                        id=self.instance_id, page=page, page_size=get_findings_page_size
+                    )
+                )
+                all_findings.extend(resp.findings)
+                if (
+                    len(resp.findings) < get_findings_page_size
+                    or not resp.findings
```

Comment:
if `resp.findings` is an empty list, the length will be zero, but if you do `len(None)` then you will get an exception, so I think we should change the order, first if `resp.findings` is not truthy and then the length check